src/main/java/Duke.java

```diff
@@ -1,10 +1,267 @@
+
+        File dir;
+        File f;
+        ArrayList<Task> list = new ArrayList<>();
+
+        dir = new File("./data");
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+        f = new File(dir, "duke.txt");
+        if (!f.exists()) {
+            try {
+                f.createNewFile();
+            } catch (IOException ex) {
+                System.out.println(ex.toString());
+            }
+        }
+        try {
+            list = readFileContents(f);
+        } catch (FileNotFoundException fe) {
+            System.out.println("File corrupted! Failed to create new file.");
+        }
+
+        String line = "_____________________________________________";
+        String greeting = "Hello! I'm Donald.\nWhat can I do for you?\n";
+        System.out.println(line);
+        System.out.println(greeting + line + "\n");
+
+        Scanner sc = new Scanner(System.in);
+        String input;
+        int index;
+        Task t;
+
+        while (sc.hasNextLine()) {
+            input = sc.nextLine();
+            System.out.println(line);
+            if (input.equals("bye")) {
+                break;
+            } else if (input.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                index = 1;
+                for (Task s : list) {
+                    System.out.println(index + "." + s);
+                    index++;
+                }
+            } else {
+                try {
+                    processInput(input, list);
+                } catch (DukeException de) {
+                    System.out.println(de.toString());
+                } catch (NumberFormatException nfe) {
+                    System.out.println("DukeException: Please input valid integer.");
+                }
+            }
+            System.out.println(line + "\n");
+        }
+
+        try {
+            FileWriter fw = new FileWriter(f);
+            for (Task task: list) {
+                fw.write(task.toSave() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Something went wrong.. Failed to save list.");
+        }
+
+        String closing = "Bye. Hope to see you again soon!\n";
+        System.out.println(closing + line);
+    }
+
+    private static ArrayList<Task> readFileContents(File filePath) throws FileNotFoundException {
+        ArrayList<Task> list = new ArrayList<>();
+        Scanner sc = new Scanner(filePath);
+        String s;
+        String[] stringSplit;
+        Task t;
+        while (sc.hasNext()) {
+            s = sc.nextLine();
+            try {
+                stringSplit = s.split("-");
+                for (int i = 0; i < stringSplit.length; i++) {
+                    stringSplit[i] = stringSplit[i].trim();
+                    System.out.println(stringSplit[i]);
+                }
+                boolean marked = Boolean.valueOf(stringSplit[1]);
```

Comment:
```suggestion
                boolean isMark= Boolean.valueOf(stringSplit[1]);
```
Can consider changing the name of the boolean variable to sound like booleans.